src/luarocks/build/rust-mlua.lua

```diff
@@ -57,7 +57,7 @@ function mlua.run(rockspec, no_install)
         cmd_sep = "&&"
     end
 
-    table.insert(cmd, "cargo build --release --features " .. table.concat(features, ","))
+    table.insert(cmd, "CARGO_TARGET_DIR=./target cargo build --release --features " .. table.concat(features, ","))
```

Comment:
Regarding @khvzak's concern:

> My only concern is when the env forcely set, it would not inherit external defs.

What do we know / can we assume about the shell this command will run in? If it's a POSIX shell we can avoid overriding an existing variable while still setting it if not:

```suggestion
    table.insert(cmd, "CARGO_TARGET_DIR=${CARGO_TARGET_DIR:=./target} cargo build --release --features " .. table.concat(features, ","))
```

Reply:
But in opendal use case, we do not just want to avoid overriding it , the root case is https://github.com/khvzak/luarocks-build-rust-mlua/blob/main/src/luarocks/build/rust-mlua.lua#L74 can not make sure this path is correct

Reply:
Oh, I know your mean, That makes sense. we can not polish some env.

Reply:
But I still think we need to resolve the path problem, make `luarocks-build-rust-mlua` find correct `target` path, my PR seems like some tricks

Reply:
One *can* find the correct target path, but it's a PITA. If you have something to parse JSON available, you can query `cargo` for it:

```console
$ cargo test -q --no-run --message-format=json | jq 'select(.reason == "compiler-artifact" and .target.test and (.target.kind | index("bin"))) | .executable'
```

Obviously adjust for release or other builds and for just getting the directory if you won't want the final executable name.

Reply:
I made a PR with a proposed fix: https://github.com/khvzak/luarocks-build-rust-mlua/pull/2

Instead of rewriting `CARGO_TARGET_DIR`, can you set the correct path in `rockspec` file?

Reply:
Thanks @khvzak @alerque !